protocol/wasmbinding/msg_plugin.go

```diff
@@ -3,136 +3,112 @@ package wasmbinding
 import (
 	"encoding/json"
 
-	errorsmod "cosmossdk.io/errors"
-	wasmkeeper "github.com/CosmWasm/wasmd/x/wasm/keeper"
 	wasmvmtypes "github.com/CosmWasm/wasmvm/types"
 	sdk "github.com/cosmos/cosmos-sdk/types"
 
 	bindings "github.com/dydxprotocol/v4-chain/protocol/wasmbinding/bindings"
 
-	sendingkeeper "github.com/dydxprotocol/v4-chain/protocol/x/sending/keeper"
-	sendingtypes "github.com/dydxprotocol/v4-chain/protocol/x/sending/types"
-
-	clobkeeper "github.com/dydxprotocol/v4-chain/protocol/x/clob/keeper"
 	clobtypes "github.com/dydxprotocol/v4-chain/protocol/x/clob/types"
+	sendingtypes "github.com/dydxprotocol/v4-chain/protocol/x/sending/types"
+	"github.com/dydxprotocol/v4-chain/protocol/x/subaccounts/types"
 )
 
-// TODO(OTE-409): add checks for contract addresses
-
-// CustomMessageDecorator returns decorator for custom CosmWasm bindings messages
-func CustomMessageDecorator(sending *sendingkeeper.Keeper, clob *clobkeeper.Keeper) func(wasmkeeper.Messenger) wasmkeeper.Messenger {
-	return func(old wasmkeeper.Messenger) wasmkeeper.Messenger {
-		return &CustomMessenger{
-			wrapped: old,
-			sending: sending,
-			clob:    clob,
-		}
-	}
-}
-
-type CustomMessenger struct {
-	wrapped wasmkeeper.Messenger
-	sending *sendingkeeper.Keeper
-	clob    *clobkeeper.Keeper
-}
-
-var _ wasmkeeper.Messenger = (*CustomMessenger)(nil)
-
 // DispatchMsg executes on the contractMsg.
-func (m *CustomMessenger) DispatchMsg(ctx sdk.Context, contractAddr sdk.AccAddress, contractIBCPortID string, msg wasmvmtypes.CosmosMsg) ([]sdk.Event, [][]byte, error) {
-	if msg.Custom != nil {
-		// only handle the happy path where this is really creating / minting / swapping ...
-		// leave everything else for the wrapped version
-		var contractMsg bindings.SendingMsg
-		// print the custom message in string format
-		if err := json.Unmarshal(msg.Custom, &contractMsg); err != nil {
-			return nil, nil, errorsmod.Wrap(err, "Error Unmarshalling Custom Message")
-		}
-		if contractMsg.CreateTransfer != nil {
-			return m.createTransfer(ctx, contractAddr, contractMsg.CreateTransfer)
-		}
-		if contractMsg.DepositToSubaccount != nil {
-			return m.depositToSubaccount(ctx, contractAddr, contractMsg.DepositToSubaccount)
-		}
-		if contractMsg.WithdrawFromSubaccount != nil {
-			return m.withdrawFromSubaccount(ctx, contractAddr, contractMsg.WithdrawFromSubaccount)
-		}
-		if contractMsg.PlaceOrder != nil {
-			return m.placeOrder(ctx, contractAddr, contractMsg.PlaceOrder)
-		}
-		if contractMsg.CancelOrder != nil {
-			return m.cancelOrder(ctx, contractAddr, contractMsg.CancelOrder)
-		}
-		return nil, nil, wasmvmtypes.InvalidRequest{Err: "Unknown custom message"}
+func CustomEncoder(sender sdk.AccAddress, msg json.RawMessage) ([]sdk.Msg, error) {
```

Comment:
Nit: rename to `EncodeDydxCustomWasmMessage` for parallelism with [native encoders](https://github.com/CosmWasm/wasmd/blob/5ae2f4a1980ba7541e5fa8ac047914fd5fe1955e/x/wasm/keeper/handler_plugin_encoders.go#L113)

Reply:
Also, can we document that `sender` here is enforced by `wasmd` to be the `contractAddr`, and point to relevant code [here](https://github.com/CosmWasm/wasmd/blob/5ae2f4a1980ba7541e5fa8ac047914fd5fe1955e/x/wasm/keeper/handler_plugin_encoders.go#L96)

Reply:
Did we address the 2nd comment above?

Reply:
https://github.com/dydxprotocol/v4-chain/pull/1809/files#diff-2b622a69eb86b29c567dd290dfe1d885dfe6f37e6af81b0a1c212ca9d6727815R49-R50

Made the change here